pkgs/applications/window-managers/hyprland/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, libdrm
+, libinput
+, libxcb
+, libxkbcommon
+, mesa
+, pango
+, wayland
+, wayland-protocols
+, wlroots
+, xcbutilwm
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hyprland";
+  version = "unstable-23-04-2022";
+
+  src = fetchFromGitHub {
+    owner = "vaxerski";
+    repo = pname;
+    rev = "fa38dfd416b664033a96bb4af764fd54f7257d7b";
+    sha256 = "sha256-wP0I8Y2cC6XadL5+XEXJnIKlPDo6u7gJys0TykqBe4A=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+
+    wayland
```

Comment:
One can also use `wayland-scanner` here and `wayland` in `buildInputs` to make it more clear. I'd also drop the empty line above. But I'd say both changes are optional as it works either way (although the first change has advantages).